catalog/resource_grants.go

```diff
@@ -97,13 +97,20 @@ func NewPermissionsAPI(ctx context.Context, m any) PermissionsAPI {
 	return PermissionsAPI{m.(*common.DatabricksClient), context.WithValue(ctx, common.Api, common.API_2_1)}
 }
 
+func getPermissionEndpoint(securable, name string) string {
+	if securable == "share" {
+		return fmt.Sprintf("/unity-catalog/shares/%s/permissions", name)
+	}
+	return fmt.Sprintf("/unity-catalog/permissions/%s/%s", securable, name)
+}
+
```

Comment:
@wchau why did we design API in such an inconsistent way, so that we have to add ifs in the client code? :)

Reply:
We have an internal ticket to consolidate these APIs. The difficulty lies in the fact that share permissions are granted to recipients, instead of a "principal". We would have to introduce a new recipient principal type.